functions/api/dashboard.js

```diff
@@ -0,0 +1,124 @@
+const constants = require('../common/constants')
+const { getInventory } = require('./inventory')
+const admin = require('firebase-admin');
+const express = require('express');
+const router = express.Router();
+const db = admin.firestore();
+
+/**
+ * @description Route to retireve all data for the dashboard from firestore
+ * If the user is Admin or SuperAdmin return all dashboard data from all Branches
+ * If the user is Branch Userreturn dashboard data from that specific branch
+ * @returns Json object containing all metadata
+ */
+router.get("/", async (request, response, next) => {
+
+    const userRole = request.user.role
```

Comment:
i thought vinayak wanted role as query param. Admin can see entire view or just branch view by passing role?

Reply:
No, this role is set after the user is authenticated in the auth, right? I am just reusing that.